packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapMarkerController.m

```diff
@@ -247,37 +278,75 @@ - (instancetype)initWithMethodChannel:(FlutterMethodChannel *)methodChannel
-    [controller interpretMarkerOptions:marker registrar:self.registrar];
-    self.markerIdentifierToController[identifier] = controller;
+    [self addMarker:marker];
   }
 }
 
+- (void)addMarker:(NSDictionary *)markerToAdd {
+  NSString *identifier = markerToAdd[@"markerId"];
+  NSString *clusterManagerId = markerToAdd[@"clusterManagerId"];
+  CLLocationCoordinate2D position = [FLTMarkersController getPosition:markerToAdd];
+  GMSMarker *marker = [GMSMarker markerWithPosition:position];
+  FLTGoogleMapMarkerController *controller =
+      [[FLTGoogleMapMarkerController alloc] initWithMarker:marker
+                                                identifier:identifier
+                                          clusterManagerId:clusterManagerId
+                                                   mapView:self.mapView];
+  [controller interpretMarkerOptions:markerToAdd registrar:self.registrar];
+  if (clusterManagerId && clusterManagerId != (id)[NSNull null]) {
+    GMUClusterManager *clusterManager =
+        [_clusterManagersController getClusterManagerWithIdentifier:clusterManagerId];
+    if (marker && clusterManager != (id)[NSNull null]) {
+      [clusterManager addItem:(id<GMUClusterItem>)marker];
+    }
+  }
+  self.markerIdentifierToController[identifier] = controller;
+}
+
 - (void)changeMarkers:(NSArray *)markersToChange {
   for (NSDictionary *marker in markersToChange) {
-    NSString *identifier = marker[@"markerId"];
-    FLTGoogleMapMarkerController *controller = self.markerIdentifierToController[identifier];
-    if (!controller) {
-      continue;
-    }
-    [controller interpretMarkerOptions:marker registrar:self.registrar];
+    [self changeMarker:marker];
+  }
+}
+
+- (void)changeMarker:(NSDictionary *)markerToChange {
+  NSString *identifier = markerToChange[@"markerId"];
+  NSString *clusterManagerId = markerToChange[@"clusterManagerId"];
+  FLTGoogleMapMarkerController *controller = self.markerIdentifierToController[identifier];
+  if (!controller) {
+    return;
+  }
+  NSString *oldClusterManagerId = [controller clusterManagerId];
+  if (![oldClusterManagerId isEqualToString:clusterManagerId]) {
+    [self removeMarker:identifier];
+    [self addMarker:markerToChange];
+  } else {
+    [controller interpretMarkerOptions:markerToChange registrar:self.registrar];
   }
 }
 
 - (void)removeMarkersWithIdentifiers:(NSArray *)identifiers {
   for (NSString *identifier in identifiers) {
-    FLTGoogleMapMarkerController *controller = self.markerIdentifierToController[identifier];
-    if (!controller) {
-      continue;
+    [self removeMarker:identifier];
+  }
+}
+
+- (void)removeMarker:(NSString *)identifier {
+  FLTGoogleMapMarkerController *controller = self.markerIdentifierToController[identifier];
+  if (!controller) {
+    return;
+  }
+  NSString *clusterManagerId = [controller clusterManagerId];
+  if (clusterManagerId && clusterManagerId != (id)[NSNull null]) {
```

Comment:
Same casting issue. If `clusterManagerId` is returning `NSNull` it is doing so incorrectly.